src/utils/scene-graph.js

```diff
@@ -4,3 +4,21 @@ export function findAncestorWithComponent(entity, componentName) {
   }
   return entity;
 }
+
+export function findComponentsInNearestAncestor(entity, componentName) {
+  const components = [];
+  while (entity) {
+    if (entity.components) {
+      for (const c in entity.components) {
+        if (c.startsWith(componentName)) {
```

Comment:
Would not have guessed in the code above using this that it was using a prefix search, though I guess that is the only way it would return multiple components from a single entity given a single component name... But then I am pretty confused by usage to find "position-at-border"

Reply:
Ah.. I forgot how aframe handles multiple instances of the same component with a prefix... I think there is a component property that actually has the component name, so you can avoid the `startsWith` check which I am sure will bite us at some point.